File: frontend/src/api/aiConfig.ts
import { requestJson } from './http'

export interface AIConfig {
  has_api_key: boolean
  base_url: string
  model: string
  updated_at: number
}

export interface AIConfigInput {
  api_key?: string
  base_url: string
  model: string
}

export interface ConnectionTestResult {
  success: boolean
  status_code: number | null
  duration_ms: number
  message: string
  detail: string
}

const getConfig = async () => {
  const response = await requestJson<AIConfig>('/ai/config')
  return response.data
}

const saveConfig = async (payload: AIConfigInput) => {
  const response = await requestJson<AIConfig>('/ai/config', {
    method: 'PUT',
    body: JSON.stringify(payload)
  })
  return response.data
}

const testBackendConnection = async (payload: AIConfigInput) => {
  const response = await requestJson<ConnectionTestResult>('/ai/config/test', {
    method: 'POST',
    body: JSON.stringify(payload)
  })
  return response.data
}

export {
  getConfig,
  saveConfig,
  testBackendConnection
}

File: backend/app/models/ai_config.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class AIConfig:
    id: str
    api_key: str
    base_url: str
    model: str
    created_at: float
    updated_at: float

    @classmethod
    def from_dict(cls, data):
        return cls(
            id=str(data.get("id", "")),
            api_key=data.get("api_key", ""),
            base_url=data.get("base_url", ""),
            model=data.get("model", ""),
            created_at=data.get("created_at", 0.0),
            updated_at=data.get("updated_at", 0.0),
        )

    def to_dict(self):
        return {
            "id": self.id,
            "api_key": self.api_key,
            "base_url": self.base_url,
            "model": self.model,
            "created_at": self.created_at,
            "updated_at": self.updated_at,
        }